src/reducers/shelf/filter.ts

```diff
@@ -19,7 +19,8 @@ export function filterReducer(shelfSpec: Readonly<ShelfUnitSpec> = DEFAULT_SHELF
       const {filter} = action.payload;
       let index = action.payload.index;
       if (contains(shelfSpec.filters, filter)) {
-        throw new Error('Cannot add more than one filter of the same field');
+        window.alert('Cannot add more than one filter of the same field');
```

Comment:
Reading more, throwing error in reducer is not a good idea either.

Instead, we may want to move the validation logic to the relevant filter components